app/domain/entities/enrolment_request.py

```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional
+
+from pydantic import BaseModel, Field
+
+
+class State(str, Enum):
+    NEW = "new"
+    INFO_REQUESTED = "info_requested"
+    ACCEPTED = "accepted"
+    REJECTED = "rejected"
+    CANCELLED = "cancelled"
+    WITHDRAWN = "withdrawn"
+    COMPLETED = "completed"
+
+
+VALID_STATE_CHANGES = (
+    (State.NEW, State.ACCEPTED),
+    (State.NEW, State.REJECTED),
+    (State.NEW, State.INFO_REQUESTED),
+    (State.INFO_REQUESTED, State.ACCEPTED),
+    (State.INFO_REQUESTED, State.REJECTED),
+    (State.INFO_REQUESTED, State.INFO_REQUESTED),
+    (State.ACCEPTED, State.CANCELLED),
+    (State.ACCEPTED, State.WITHDRAWN),
+    (State.ACCEPTED, State.COMPLETED),
+)
+
+
+class EnrolmentRequest(BaseModel):
```

Comment:
Shouldn't it be same to enrolment request structure in TP-Enrolments?
with course_id, student_id etc. use ?
https://github.com/ACWIC/training-provider-enrolment/blob/main/app/domain/entities/enrolment_authorisation.py

it should be status/state?
let me know if you want me to update it in TP-Enrolments